### PR TITLE
feat(Radio): component boilerplate

### LINKED ISSUE #241: [Feat]: New Radio component
**Description**:

The `Radio `component allows users to select a single option from a set of predefined options. It is ideal for use in forms, settings pages, or any context where only one choice from a group is needed.

**Requirements**:

1. **Component Implementation:**

- Implement the `Radio `component in TypeScript.
- Utilize SCSS with BEM methodology for styling.
- Integrate design tokens from @geneui/tokens for consistent colors, font sizes, spacing, and borders.
- Support different configurations such as disabled state, custom labels, and group functionality.
- Ensure accessibility with proper ARIA roles and keyboard navigation support.

2. **Unit Tests:**

- Write unit tests using Jest and React Testing Library.
- Ensure test coverage for all props, including rendering of different states (e.g., checked, unchecked, disabled).
- Test for edge cases such as multiple unchecked radios, group functionality, and proper interaction.

3. **Storybook Stories:**

- Create a Storybook story for the Radio component.
- Showcase different configurations, such as checked, unchecked, and disabled states.
- Create controls for the props so users can interact with the Radio component in Storybook.
- Document all props of the public interfaces.

4. **Figma Design:**

Follow the design outlined in the provided [Figma link.](https://www.figma.com/design/XmW63z2fXfHBC95eCM36ed/Gene-UI%3A-Components?node-id=141-43&p=f&m=dev)

**Checklist**:

 

- [ ] Component is implemented in TypeScript.
- [ ]  Styling follows SCSS using BEM methodology.
- [ ]  Design tokens from @geneui/tokens are used for colors, font sizes, spacing, and borders.
- [ ]  Unit tests written and cover different states and edge cases.
- [ ]  Storybook stories created with examples of all props and configuration options.
- [ ]  Figma design matches the component implementation.

### PR DESCRIPTION
**Pull request by NarekArshakyan**
### Chromatic Build Summary 
**Build URL:** https://www.chromatic.com/build?appId=66daee7859a843dd87bfb3d7&number=386 
**Storybook URL:** https://66daee7859a843dd87bfb3d7-obdibrfytv.chromatic.com/

### CODE REVIEW

File: src/components/atoms/Radio/Radio.tsx [hunk 1]
```diff
@@ -0,0 +1,182 @@
+import React, { ChangeEvent, FC, useEffect, useRef, FocusEvent, useState } from "react";
+import { CircleFilled } from "@geneui/icons";
+import classNames from "classnames";
+
+// Components
+import Label from "../../atoms/Label";
```
Comment (hamikhambardzumyan) [inline src/components/atoms/Radio/Radio.tsx]: Use alias imports for components

File: src/components/atoms/Radio/Radio.tsx [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+interface IRadioProps {
+    /**
+     *  The text displayed as the label for the radio, describing its purpose or function.
+     */
+    label?: string;
+    /**
+     *  Toggles the label's and HelperText position between above or beside the radio.
+     */
+    vertical?: boolean;
```
Comment (hamikhambardzumyan) [inline src/components/atoms/Radio/Radio.tsx]: Rename this prop to `direction` with a default `horizontal` value.

File: src/components/atoms/Radio/Radio.tsx [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+    /**
+     *  Manages the checked state of the radio in a controlled way.
+     */
+    checked?: boolean;
+    /**
+     *  Automatically focuses the radio when the component mounts.
+     */
+    autoFocus?: boolean;
+    /**
+     *  Extra information displayed with the label for clarity or guidance.
```
Comment (hamikhambardzumyan) [inline src/components/atoms/Radio/Radio.tsx]:
```suggestion
     *  Extra information displayed with the tooltip for clarity or guidance.
```

File: src/components/atoms/Radio/Radio.tsx [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+     */
+    infoText?: string;
+    /**
+     *  Helper text to provide context or explain any errors, warnings related to the radio.
+     */
+    helperText?: string;
+    /**
+     *  The initial checked state of the radio before user interaction.
```
Comment (hamikhambardzumyan) [inline src/components/atoms/Radio/Radio.tsx]:
```suggestion
     *  The initial state of the radio was checked before user interaction. This prop does not make the component controlled.
```

File: src/components/atoms/Radio/Radio.tsx [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+     *  Event handler for when the radio input element receives focus. Provides the focus event as a callback's argument.
+     */
+    onFocus?: (e: FocusEvent<HTMLInputElement>) => void;
+    /**
+     * Additional class for the parent element.
+     * This prop should be used to set placement properties for the element relative to its parent using BEM conventions.
+     */
+    className?: string;
+    /**
+     * The value that returns in the onChange event
```
Comment (hamikhambardzumyan) [inline src/components/atoms/Radio/Radio.tsx]:
```suggestion
     * The value of the component that will be returned in the onChange event.
```

File: src/components/atoms/Radio/Radio.tsx [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+     * */
+    value: string;
+}
+
+/**
+ * radio component allows users to select one or more options from a set of choices. Each radio can be either checked or unchecked, indicating a binary state. radioes are commonly used in forms, settings, and lists where multiple selections are needed.
```
Comment (hamikhambardzumyan) [inline src/components/atoms/Radio/Radio.tsx]:
```suggestion
 * The radio component allows users to select one or more options from a set of choices. Each radio can be either checked or unchecked, indicating a binary state. Radios are commonly used in forms, settings, and lists where a user needs to select a single option from multiple options.
```

File: src/components/atoms/Radio/Radio.tsx [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+ */
+const Radio: FC<IRadioProps> = ({
+    label,
+    required,
+    infoText,
+    disabled,
+    helperText,
+    readOnly,
+    type = "rest" as const,
```
Comment (hamikhambardzumyan) [inline src/components/atoms/Radio/Radio.tsx]: Please replace the code to 

```suggestion
    type = "rest",
```

File: src/components/atoms/Radio/Radio.tsx [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+    onFocus,
+    onBlur,
+    name,
+    checked,
+    defaultChecked,
+    className,
+    value
+}) => {
+    const interRef = useRef<HTMLInputElement | null>(null);
+    const isControlled = checked !== undefined;
```
Comment (hamikhambardzumyan) [inline src/components/atoms/Radio/Radio.tsx]: For most React “controlled vs. uncontrolled” decisions, 'checked' in props is the more robust choice. It directly corresponds to “did the parent provide this prop at all?” and lines up with React’s own concept of controlled inputs. If the key is present, consider it controlled; if it’s absent, consider it uncontrolled.

Replace restructuring props to props in the component argument and restructure all props inside of the component body.

```suggestion
    const isControlled = 'checked' in props;
```

File: src/components/atoms/Radio/Radio.tsx [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+    onChange,
+    onFocus,
+    onBlur,
+    name,
+    checked,
+    defaultChecked,
+    className,
+    value
+}) => {
+    const interRef = useRef<HTMLInputElement | null>(null);
```
Comment (hamikhambardzumyan) [inline src/components/atoms/Radio/Radio.tsx]: Why do you need this ref?

**Review by hamikhambardzumyan — CHANGES_REQUESTED** [review]

**Review by hamikhambardzumyan — APPROVED** [review]
Looks good to me
